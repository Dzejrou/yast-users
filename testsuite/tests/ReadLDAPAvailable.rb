# encoding: utf-8

# File		: ReadLDAPAvailable.ycp
# Module	: Users configurator
# Summary	: Test of UsersLDAP::ReadAvailable
# Authors	: Jiri Suchomel <jsuchome@suse.cz>
module Yast
  class ReadLDAPAvailableClient < Client
    def main
      # testedfiles: UsersLDAP.pm

      @READ = { "etc" => { "nsswitch_conf" => { "passwd" => "ldap files" } } }

      Yast.import "Testsuite"
      Yast.import "UsersLDAP"

      Testsuite.Test(lambda { UsersLDAP.ReadAvailable }, [@READ, {}, {}], 0)

      Ops.set(@READ, ["etc", "nsswitch_conf", "passwd"], "files")

      Testsuite.Test(lambda { UsersLDAP.ReadAvailable }, [@READ, {}, {}], 0)

      Ops.set(@READ, ["etc", "nsswitch_conf", "passwd"], "sss")

      Testsuite.Test(lambda { UsersLDAP.ReadAvailable }, [@READ, {}, {}], 0)

      nil
    end
  end
end

Yast::ReadLDAPAvailableClient.new.main
