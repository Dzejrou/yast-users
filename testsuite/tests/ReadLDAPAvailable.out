Read	.ldap_conf.v.binddn 0
Return	true
Read	.ldap_conf.v.binddn 0
Return	true
Read	.ldap_conf.v.binddn 0
Return	true
